src/home_robot/home_robot/utils/simple_env.py

```diff
@@ -5,25 +5,80 @@
 from abc import ABC, abstractmethod
 from typing import Any, Dict, Optional
 
+import matplotlib.pyplot as plt
 import numpy as np
 
 from home_robot.core.abstract_env import Env
 from home_robot.core.interfaces import Action, Observations
+from home_robot.motion.space import ConfigurationSpace
 
 
 class SimpleEnv(Env):
     """Simple 2D environment for testing."""
 
-    def __init__(self, seed: int = 0):
-        self.reset(seed)
+    def __init__(self, obs: np.ndarray = None, size=10, obstacle_size=6, seed: int = 0):
+        """Create simple 2d env. obs is obstacle location."""
+        self.size = size
+        self.obstacle_size = obstacle_size
+        if obs is None:
+            self.reset(seed)
+        else:
+            self.obstacle_pos = obs
+
+    def get_space(self) -> ConfigurationSpace:
+        """Get a space that we can use for planning"""
+        return ConfigurationSpace(2, np.zeros(2), np.ones(2) * self.size, step_size=0.5)
+
+    def validate(self, q):
+        """Check to see if this configuration is feasible for planning"""
+        assert len(q) == 2
+        x, y = q
+        if x < 0 or y < 0 or x > self.size or y > self.size:
+            # Out of bounds
+            return False
+        ox, oy = self.obstacle_pos
+        if (
+            x > ox
+            and y > oy
+            and x < ox + self.obstacle_size
+            and y < oy + self.obstacle_size
+        ):
+            # inside the obstacle
+            return False
+        return True
 
-    @abstractmethod
     def reset(self, seed: int = 0):
         self.seed = seed
         np.random.seed(seed)
-        # TODO: regenerate start, goal and obstacles
+        self.obstacle_pos = np.random.random(2) * (self.size - self.obstacle_size)
+
+    def show(self, states):
```

Comment:
Can we start to implement backends, so
`def show(self, states, backend='mpl', show=True)`
Then the function by default returns a figure and only calls plt.show() if show == True

Reply:
done